### PR TITLE
Fixed inital compound error

### LINKED ISSUE #1311: Fix the initial compounds not reacting to the organelles
https://community.revolutionarygamesstudio.com/t/general-0-5-0-feedback/2645/9?u=hhyyrylainen

### PR DESCRIPTION
**Pull request by CoreyHendrey**
I added a new function to MicrobeSpecies.cs to fix this problem.

The compounds were not being updated after you edited your microbe. 

Closes #1311

### CODE REVIEW

File: src/general/Mutations.cs [hunk 1]
```diff
@@ -125,24 +125,7 @@ public MicrobeSpecies CreateMutatedSpecies(MicrobeSpecies parent, MicrobeSpecies
...
-        else if (mutated.Organelles.Any(o => o.Definition == chemo ||
-            o.Definition == chemoProtein))
-        {
-            mutated.SetInitialCompoundsForChemo();
-        }
-        else
-        {
-            mutated.SetInitialCompoundsForDefault();
-        }
+        mutated.SetUpdatedCompounds();
```
Comment (hhyyrylainen) [inline src/general/Mutations.cs:128]: I think this should be named `UpdateInitialCompounds`

File: src/microbe_stage/MicrobeSpecies.cs [hunk 2]
```diff
@@ -51,6 +53,30 @@ public void SetInitialCompoundsForChemo()
         InitialCompounds.Add("hydrogensulfide", 10);
     }
 
+    public void SetUpdatedCompounds()
+    {
+        var simulation = SimulationParameters.Instance;
+
+        // If you have iron (f is the symbol for rusticyanin)
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeSpecies.cs:60]: I don't think this comment is accurate or really adds anything, should be just removed.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -363,6 +363,8 @@ public void OnFinishEditing()
         parent.AddChild(ReturnToStage);
         ReturnToStage.OnReturnFromEditor();
 
+        editedSpecies.UpdateInitialCompounds();
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:366]: I think this should be put after line 325, as it makes more sense to be there, than after performing the scene switch.

Comment (CoreyHendrey) [reply src/microbe_stage/editor/MicrobeEditor.cs]: Yep, moved.

**Pull request merged**